File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
 float amount=0.0;
 int cents=0;
 int quarters=0;
 int dimes=0;
 int nickels=0;
 int leftover=0;
 do{
   printf("O hai! How much change is owed? ");
   amount = get_float();
 } while(amount <= 0 || amount == 0);
 cents = (int)round(amount*100.0);
 
 //check quarters
 quarters = cents / 25;
 leftover = cents % 25;
 
 //check dimes
 dimes = leftover / 10;
 leftover = leftover % 10;
 
 //check nickels
 nickels = leftover / 5;
 leftover = leftover % 5;

 
 printf("%d\n",quarters+dimes+nickels+leftover);
 
 




  return 0;
}

File: pset6/greedy.py
import cs50

amount = 0.0
cents = 0
quarters = 0
dimes = 0
nickels = 0
leftover = 0

print("O hai! How much change is owed? ", end="")
amount = cs50.get_float()
if amount <= 0 or amount == 0:
    while amount <= 0 or amount == 0:
        print("O hai! How much change is owed? ", end="")
        amount = cs50.get_float()  
        
cents = amount*int(100)

quarters = int(cents) // 25
leftover = int(cents) % int(25)


dimes = int(leftover) // int(10)
leftover = int(leftover) % int(10)


nickels = int(leftover) // int(5)
leftover = int(leftover) % int(5)


print("{}".format(quarters+dimes+nickels+leftover))
    